pkg/kubectl/apply.go

```diff
@@ -186,3 +191,27 @@ func CreateOrUpdateAnnotation(createAnnotation bool, info *resource.Info, codec
 	}
 	return UpdateApplyAnnotation(info, codec)
 }
+
+func GetModifiedConfigurationWithNulls(info *resource.Info, codec runtime.Encoder) ([]byte, error) {
+	modified, err := GetModifiedConfiguration(info, true, codec)
+	if err != nil {
+		return nil, err
+	}
+
+	// During Unmarshaling of the user's object config to a Go struct, explicit null values are not preserved.
+	// For this, we merge the raw configuration file (that has null values) with the currently loaded
+	// configuration (which has extra information like annotations and empty structs). There might be conflicts,
+	// which we resolve in favor of info.Raw. The conflicts should only be empty initialized structs that we will
+	// overwrite only if the user has defined a null.
+	// TODO(andronat): Do we want to store user's explicit null values?
+	tObj := info.VersionedObject
+	if tObj == nil {
+		tObj = info.Object
+	}
+	modified, err = strategicpatch.GuidedJsonMerge(modified, info.Raw, tObj)
```

Comment:
This is not safe and you cannot assume Raw is JSON.  You're going to need to be more sophisticated here.